runtime/databricks/automl_runtime/forecast/utils.py

```diff
@@ -111,3 +112,57 @@ def generate_cutoffs(df: pd.DataFrame, horizon: int, unit: str,
 
 def is_quaterly_alias(freq: str):
     return freq in QUATERLY_OFFSET_ALIAS
+
+def is_frequency_consistency(
+                start_time: Union[pd.Series, pd.Timestamp],
+                end_time: Union[pd.Series, pd.Timestamp], 
+                freq:str) -> bool:
+    """
+    Validate the periods given a start time, end time is consistent with given frequency.
+    :param start_time: A pd series convertable to datetime
+    :param end_time: A pd series convertable to datetime, must be in same size
+                as start_time.
+    :param freq: A string that is accepted by OFFSET_ALIAS_MAP, e.g. 'day',
+                'month' etc.
+    :return: A boolean indicate whether the time interval is
+             evenly divisible by the period.
+    """
+    periods = calculate_periods(start_time, end_time, freq)
+    diff = pd.to_datetime(end_time) -  pd.DateOffset(
+                **DATE_OFFSET_KEYWORD_MAP[OFFSET_ALIAS_MAP[freq]]
+            ) * periods == pd.to_datetime(start_time)
+    if type(diff) is bool:
+        return diff
+    return diff.all()
+
+
+def calculate_periods(
+                start_time: Union[pd.Series, pd.Timestamp],
+                end_time: Union[pd.Series, pd.Timestamp], 
```

Comment:
Similarly, why do we need start_time (and end_time) can be both a pd.Series and a pd.Timestamp? I felt the code'll be easier to read if we can make the type more specific...

Also this function is actually calculating the period difference but not periods (the pandas's PeriodIndex). Could we rename it to be something like `calculate_period_difference`?

Reply:
Done.